new: tutorial - User Profile with Flow UnitySDK (#75)
emerald-dao/emerald-academy-v2: src/lib/content/tutorials/flow-unity-sdk/en/readme.md

---

> title: User Profile with Flow UnitySDK

Reviewer: ```suggestion
title: User Profile with Flow Unity SDK
```

lesson: 1
language: en

> excerpt: User Profile with Flow UnitySDK

Reviewer: ```suggestion
excerpt: User Profile with Flow Unity SDK
```

---

> # Tutorial - Using Flow UnitySDK

Reviewer: ```suggestion
# Tutorial - Using Flow Unity SDK
```

> GM! Welcome to another beautiful day of Cadence development! I'm Memxor and today we will learn how to use the Flow UnitySDK. 

Reviewer: ```suggestion
GM! Welcome to another beautiful day of Cadence development! I'm Memxor and today we will learn how to use the Flow Unity SDK. 
```

In this tutorial we will be using the [User Profile Contract](https://academy.ecdao.org/en/catalog/tutorials/user-profile) tutorial. If you haven't gone through it, please do so we are on the same page. We will be implementing the same User Profile that we maded in Cadence into Unity.

## Step 1 - Setting up the Environemnt

> 1. Go to Unity Asset Store and add the [Flow UnitySDK](https://assetstore.unity.com/packages/decentralization/flow-sdk-237172) to your account.

Reviewer: ```suggestion
1. Go to Unity Asset Store and add the [Flow Unity SDK](https://assetstore.unity.com/packages/decentralization/flow-sdk-237172) to your account.
```

![Unity UI](https://i.imgur.com/UHZtPwq.png)

Now, let's make a script to handle all of our functions, I'll name it `FlowHandler.cs` - 

```cs
using System.Threading.Tasks;
using DapperLabs.Flow.Sdk;
using DapperLabs.Flow.Sdk.Crypto;
using DapperLabs.Flow.Sdk.DataObjects;
using DapperLabs.Flow.Sdk.Unity;
using DapperLabs.Flow.Sdk.WalletConnect;
using UnityEngine;

public class FlowHandler : MonoBehaviour
{
    public GameObject qrCodeCustomPrefab;
    public GameObject walletSelectCustomPrefab;
    private FlowControl.Account scriptsExecutionAccount;
    private FlowAccount flowAccount;
    private string walletAddress;

    private async void Start() 
    {
        FlowConfig flowConfig = new()
        {
            NetworkUrl = FlowConfig.TESTNETURL,
            Protocol = FlowConfig.NetworkProtocol.HTTP
        };

        FlowSDK.Init(flowConfig);
        IWallet walletProvider = new WalletConnectProvider();

        walletProvider.Init(new WalletConnectConfig
        {
            ProjectId = "YOUR_WALLETCONNECT_ID",

>             ProjectName = "Flow UnitySDK Tutorial",
>             ProjectDescription = "This is an tutorial on how to use the Flow UnitySDK",

Reviewer 2: ```suggestion
            ProjectName = "Flow Unity SDK Tutorial",
            ProjectDescription = "This is an tutorial on how to use the Flow Unity SDK",
```
